atlasdb-feign/src/main/java/com/palantir/atlasdb/http/AtlasDbFeignTargetFactory.java

```diff
@@ -131,15 +133,27 @@ private AtlasDbFeignTargetFactory() {
             String userAgent) {
         return RecreatingInvocationHandler.create(
                 serverListConfigSupplier,
-                serverListConfig -> createProxyWithFailover(
-                        serverListConfig.sslConfiguration().map(sslSocketFactoryCreator),
-                        serverListConfig.proxyConfiguration().map(proxySelectorCreator),
-                        serverListConfig.servers(),
-                        type,
-                        userAgent),
+                serverListConfig -> {
+                    if (serverListConfig.hasAtLeastOneServer()) {
+                        return createProxyWithFailover(
+                                serverListConfig.sslConfiguration().map(sslSocketFactoryCreator),
+                                serverListConfig.proxyConfiguration().map(proxySelectorCreator),
+                                serverListConfig.servers(),
+                                type,
+                                userAgent);
+                    }
+                    return createProxyForZeroNodes(type);
+                },
                 type);
     }
 
+    private static <T> T createProxyForZeroNodes(Class<T> type) {
+        return Reflection.newProxy(type, (unused1, unused2, unused3) -> {
+            throw new ServiceNotAvailableException("The " + type.getSimpleName() + " is currently unavailable,"
```

Comment:
I'd probably word this error message more directly: `"The " + type.getSimpleName() + " is currently unavailable," + " because configuration contains zero servers."`

Reply:
Done, following your suggested wording.